app/src/test/java/mozilla/lockbox/presenter/AppLockedPresenterTest.kt

```diff
@@ -0,0 +1,185 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+@file:Suppress("DEPRECATION")
+
+package mozilla.lockbox.presenter
+
+import android.app.KeyguardManager
+import android.content.Context
+import android.hardware.fingerprint.FingerprintManager
+import io.reactivex.Observable
+import io.reactivex.subjects.PublishSubject
+import io.reactivex.subjects.Subject
+import junit.framework.Assert.assertEquals
+import junit.framework.Assert.assertFalse
+import junit.framework.Assert.assertTrue
+import mozilla.lockbox.action.DataStoreAction
+import mozilla.lockbox.action.FingerprintAuthAction
+import mozilla.lockbox.action.RouteAction
+import mozilla.lockbox.action.UnlockingAction
+import mozilla.lockbox.flux.Dispatcher
+import mozilla.lockbox.store.FingerprintStore
+import mozilla.lockbox.store.LockedStore
+import mozilla.lockbox.store.SettingStore
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.spy
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(packageName = "mozilla.lockbox")
+class AppLockedPresenterTest {
```

Comment:
The more logic you push into the common `LockedPresenter` the less duplication you need for tests.